lib/pdk/util/filesystem.rb

```diff
@@ -4,6 +4,9 @@ module Filesystem
       def write_file(path, content)
         raise ArgumentError unless path.is_a?(String) || path.respond_to?(:to_path)
 
+        # Make sure all written files have a trailing newline.
+        content += "\n" unless content[-1] == "\n"
```

Comment:
Perhaps we should do the `content.encode` before checking the last character? What do you folks think?

Reply:
Yeah, just for safety we should reencode first.